{% extends 'layout.html' %}

{% block header %}
  <h1>Sequencing Report: Coverage</h1>

  <table class="table">
    <tbody>
      <tr>
        <th class="table-header--vertical">Samples included</th>
        <td class="center-text">99-1-1A</td>
        <td class="center-text">101-1-1A</td>
      </tr>
    </tbody>
  </table>
{% endblock %}

{% block body %}
  <section id="key_metrics">
    <h2>Key metrics</h2>

    <table class="fb-table--equalized">
      <thead>
        <tr>
          <th>Sample Id</th>
          <th>Cutoff</th>
          <th>Avg. Coverage</th>
          <th>Avg. Completeness [%]</th>
          <th>Diagnostic Yield [%]</th>
          <th>Base coverage [%]</th>
        </tr>
      </thead>

      <tbody>
        <tr>
          <td>101-1-1A</td>
          <td>10</td>
          <td>47.7098</td>
          <td>99.8794</td>
          <td>98.5358</td>
          <td>99.8380</td>
        </tr>

        <tr>
          <td>99-1-1A</td>
          <td>10</td>
          <td>187.927</td>
          <td>98.5669</td>
          <td>91.9161</td>
          <td>98.4906</td>
        </tr>
      </tbody>
    </table>
  </section>

  <section id="superset_panel">
    <h2>Key metrics for dbCMMS v1.0</h2>

    <table class="fb-table--equalized">
      <thead>
        <tr>
          <th>Sample Id</th>

          <th>Cutoff</th>

          <th>Avg. Coverage</th>

          <th>Avg. Completeness [%]</th>

          <th>Diagnostic Yield [%]</th>
        </tr>
      </thead>

      <tbody>
        <tr>
          <td>101-1-1A</td>

          <td>10</td>

          <td>47.4553</td>

          <td>99.9928</td>

          <td>99.726</td>
        </tr>

        <tr>
          <td>99-1-1A</td>

          <td>10</td>

          <td>195.82</td>

          <td>99.3701</td>

          <td>94.3836</td>
        </tr>
      </tbody>
    </table>
  </section>

  <section id="gc_content_performance">
    <h2>GC Content Performance</h2>

    <table class="fb-table">
      <thead>
        <tr>
          <th rowspan="2">Sample Id</th>

          <th colspan="2">High GC Content</th>

          <th colspan="2">Low GC Content</th>
        </tr>

        <tr>
          <th>Avg. Coverage</th>

          <th>Avg. Completeness [%]</th>

          <th>Avg. Coverage</th>

          <th>Avg. Completeness [%]</th>
        </tr>
      </thead>

      <tbody>
        <tr>
          <td>101-1-1A</td>

          <td>42.3182</td>

          <td>100.0</td>

          <td>57.0116</td>

          <td>100.0</td>
        </tr>

        <tr>
          <td>99-1-1A</td>

          <td>81.2634</td>

          <td>94.7842</td>

          <td>237.4947</td>

          <td>100.0</td>
        </tr>
      </tbody>
    </table>
  </section>

  <section id="sex_check">
    <h2>Sex Check</h2>

    <table class="fb-table">
      <thead>
        <tr>
          <th>Sample Id</th>

          <th>Chromosome X Coverage</th>

          <th>Chromosome Y Coverage</th>

          <th>Sex Prediction</th>
        </tr>
      </thead>

      <tbody>
        <tr>
          <td>99-1-1A</td>

          <td>104.44</td>

          <td>43.0</td>

          <td>male</td>
        </tr>

        <tr>
          <td>101-1-1A</td>

          <td>25.09</td>

          <td>24.39</td>

          <td>male</td>
        </tr>
      </tbody>
    </table>
  </section>

  <section id="missed_transcripts">
    <h2>Missed Transcripts</h2>

    <table class="fb-table">
      <thead>
        <tr>
          <th>Gene Id</th>

          <th>Transcript Id</th>

          <th>Sample</th>

          <th>Completeness [%]</th>
        </tr>
      </thead>

      <tbody>
        <tr>
          <td>SDHD</td>

          <td>CCDS31678.1</td>

          <td>99-1-1A</td>

          <td class="text-right">65.4167</td>
        </tr>

        <tr>
          <td>SDHD</td>

          <td>CCDS60958.1</td>

          <td>99-1-1A</td>

          <td class="text-right">58.1967</td>
        </tr>

        <tr>
          <td>SMPD1</td>

          <td>CCDS31409.2</td>

          <td>101-1-1A</td>

          <td class="text-right">98.476</td>
        </tr>
      </tbody>
    </table>
  </section>

  <section id="furter-info" class="break-here">
    <h2>Further information</h2>

    <div class="text-box columns--3">
      <div class="info">
        <h3>General</h3>
        <p>The parts of the genome that have been analysed are all protein-coding parts of exons annotated in the CCDS database, together with all splice donor and acceptor sites (two bases in both ends of all introns).</p>
      </div>
      <div class="info">
        <h3>Key Metrics</h3>

        <p>Metrics are calculated on the interval/exon level for all but
        "diagnostic yield" which is calculated on the transcript level.</p>
      </div>

      <div class="info">
        <h3>Key metrics for Gene Panel</h3>

        <p>The estimated average coverage and completeness have been made on
        the transcript level (vs. exon for full key metrics). This will in
        all likelyhood not differ much from the more accurate exon level
        estimate but it should be noted that they can differ.</p>
      </div>

      <div class="info">
        <h3>GC Content Performace</h3>

        <p>GC content performace is estimated by determining the avereage
        coverage and completeness across the 10 genes with highest vs. the
        lowest GC content.</p>

        <p><strong>High GC content genes</strong>: 'UTF1', 'BHLHA9',
        'C20orf201', 'LRRC26', 'HES4', 'BHLHE23', 'C9orf172', 'NKX6-2', and
        'CITED4'.</p>

        <p><strong>Low GC content genes</strong>: 'DEFB114', 'NTS',
        'ANGPTL3', 'CYLC2', 'GPR22', 'SI', 'CSN3', 'KLRC4', and 'CSN1S1'.</p>
      </div>

      <div class="info">
        <h3>Sex check</h3>

        <p>Depending on the coverage of X and Y chromosomes, a simple still
        very accurate sex prediction can be made on the samples. This is a
        simple control step that will yield a warning that samples can have
        been mixed upstream.</p>
      </div>

      <div class="info">
        <h3>Missed transcripts</h3>

        <p>A missed transcript is defined as having less than or equal to 50%
        completeness. The idea is to set the
        <code>MISSED_TRANSCRIPT_CUTOFF</code> setting to a value where
        finding a variant in the remaining (uncovered) bases is so unlikely
        that it can be discarded.</p>

        <p>Note also that this 'risk' will be a factor both of the the
        <code>MISSED_TRANSCRIPT_CUTOFF</code> and the original completeness
        cutoff (default: 10x).</p>
      </div>
    </div>
  </section>
{% endblock %}

{% block scripts %}
  {{ super() }}
  <script src="//cdnjs.cloudflare.com/ajax/libs/d3/3.4.2/d3.js"></script>
  <script src="//cdnjs.cloudflare.com/ajax/libs/vega/1.3.3/vega.js"></script>
{% endblock %}
